plugins/processors/aws/ec2/ec2.go

```diff
@@ -0,0 +1,213 @@
+package ec2
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	internalaws "github.com/influxdata/telegraf/internal/config/aws"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"github.com/influxdata/telegraf/plugins/processors/aws/utils"
+)
+
+type (
+	EC2 struct {
+		Region    string `toml:"region"`
+		AccessKey string `toml:"access_key"`
+		SecretKey string `toml:"secret_key"`
+		RoleARN   string `toml:"role_arn"`
+		Profile   string `toml:"profile"`
+		Filename  string `toml:"shared_credential_file"`
+		Token     string `toml:"token"`
+
+		CacheTTL     internal.Duration `toml:"cache_ttl"`
+		MetricNames  []string          `toml:"metric_names"`
+		Id           string            `toml:"id"`
+		InstanceType bool              `toml:"instance_type"`
+		AmiId        bool              `toml:"ami_id"`
+		Tags         []string          `toml:"tags"`
+
+		client EC2Client
+	}
+
+	EC2Client interface {
+		DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error)
+	}
+
+	CachingEC2Client struct {
+		client  EC2Client
+		ttl     time.Duration
+		fetched time.Time
+		data    map[string]*ec2.DescribeInstancesOutput
+	}
+)
+
+func (e *CachingEC2Client) DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	id := *input.InstanceIds[0]
+	if e.data == nil {
+		e.data = map[string]*ec2.DescribeInstancesOutput{}
+	}
+	if e.fetched.IsZero() {
+		e.fetched = time.Now()
+	}
+	if time.Since(e.fetched) >= e.ttl {
```

Comment:
It would be nice to have a way to clean out old instances from the cache.  Perhaps if the length of the map is larger than some value?